kong/cache.lua

```diff
@@ -226,14 +245,21 @@ function _M:get_bulk(bulk, opts)
 end
 
 
-function _M:safe_set(key, value)
+function _M:safe_set(key, value, shadow_page)
```

Comment:
I'm not a super fan of boolean parameters, and I would have added two new functions (`shadow_safe_set(key, value)` and `shadow_purge()`). But this is not critical.

Reply:
I know you're not :) I have considered this before, but then I'd have to either duplicate code or make a private function that the two public functions dispatch to. Both seemed overkill, given the limited use. I did use a `SHADOW = true` constant in the calling code to make use of the shadow-page more obvious.